numba_dpex/core/kernel_interface/dispatcher.py

```diff
@@ -468,52 +522,56 @@ def __getitem__(self, args):
             global_range and local_range attributes initialized.
 
         """
-        if isinstance(args, int):
-            self._global_range = [args]
-            self._local_range = None
-        elif isinstance(args, tuple) or isinstance(args, list):
-            if len(args) == 1 and all(isinstance(v, int) for v in args):
-                self._global_range = list(args)
-                self._local_range = None
-            elif len(args) == 2:
-                gr = args[0]
-                lr = args[1]
-                if isinstance(gr, int):
-                    self._global_range = [gr]
-                elif len(gr) != 0 and all(isinstance(v, int) for v in gr):
-                    self._global_range = list(gr)
-                else:
-                    raise IllegalRangeValueError(kernel_name=self.kernel_name)
 
-                if isinstance(lr, int):
-                    self._local_range = [lr]
-                elif isinstance(lr, list) and len(lr) == 0:
-                    # deprecation warning
+        if isinstance(args, Ranges):
+            self._global_range = list(args.global_range)[::-1]
+            self._local_range = list(args.local_range)[::-1]
+        else:
+            if (
+                isinstance(args, tuple)
+                and len(args) == 2
+                and isinstance(args[0], int)
+                and isinstance(args[1], int)
+            ):
+                warn(
+                    "Ambiguous kernel launch paramters. "
+                    + "If your data have dimensions > 1, "
+                    + "include a default/empty local_range. "
+                    + "i.e. <function>[(M,N), numba_dpex.DEFAULT_LOCAL_RANGE](<params>), "
+                    + "otherwise your code might produce erroneous results.",
+                    DeprecationWarning,
+                )
+                self._global_range = [args[0]]
+                self._local_range = [args[1]]
+                return self
+
+            if not isinstance(args, Iterable):
+                args = [args]
+
+            ls = None
+            nargs = len(args)
+            # Check if the kernel enquing arguments are sane
+            if nargs < 1 or nargs > 2:
+                self._raise_invalid_kernel_enqueue_args()
```

Comment:
Directly raise `InvalidKernelLaunchArgsError` here the old `_raise_invalid_kernel_enqueue_args` has a totally wrong error message.

Reply:
Old functions are removed.